pyne/alara.py

```diff
@@ -943,3 +1116,214 @@ def _find_phsrc_dc(idc, phtn_src_dc):
+    -------
+    String represent the output zone block corresponding to the response.
+    """
+
+    # set default value for functions do not need response
+    if responses == None:
+        return ''
+    # input check
+    for response in responses:
+        if response not in ('decay_heat', 'photon_source', 'specific_activity',
+                        'alpha_heat', 'beta_heat', 'gamma_heat', 'wdr'):
+            raise ValueError('response {0} not supported.'.format(response))
+
+    start_str = "output zone\n"
+    end_str = "end"
+    code_block = ''
+    # define code block for decay_heat
+    if 'decay_heat' in responses:
+        code_block = ''.join([code_block, "      total_heat\n"])
+    # define code block for specific_activity
+    if 'specific_activity' in responses:
+        code_block = ''.join([code_block, "      specific_activity\n"])
+    # define code block for alpha_heat
+    if 'alpha_heat' in responses:
+        code_block = ''.join([code_block, "       alapha_heat\n"])
+    # define code block for beta_heat
+    if 'beta_heat' in responses:
+        code_block = "       beta_heat\n"
+    # define code block for gamma_heat
+    if 'gamma_heat' in responses:
+        code_block = ''.join([code_block, "       gamma_heat\n"])
+    # define code block for wdr
+    if 'wdr' in responses:
+        code_block= ''.join(["       wdr ", wdr_file, "\n"])
+    # define code block for photon_source
+    if 'photon_source' in responses:
+        alara_lib = get_alara_lib(alara_params)
+        code_block= ''.join([code_block, "       photon_source ", alara_lib,
+                             " phtn_src 1 2e7\n"])
+    return ''.join([start_str, code_block, end_str])
+
+def _is_data(line):
+    """
+    This function is used to check whether a line of alara output file contains
+    wanted data. The line contains data is conposed of:
+        - nuc : nuc name (total or TOTAL)
+        - data for each decay time (including 'shutdown': floats
+
+    Parameters
+    ----------
+    line : string
+        A line from ALARA output.txt
+
+    Returns
+    -------
+    True : if this line contains results data
+    False : if this line doesn't contain results data
+    """
+    # check the list from the second value, if they are float, then return True
+    ls = line.strip().split()
+    if len(ls) < 2:
+        return False
+    try:
+        np.array(ls[1:]).astype(float)
+        return True
+    except:
+        return False
+    
+
+def read_decay_times(line):
+    """
+    This function reads a line contian decay times information from alara
+    output file and return the decay times list.
+
+    Parameters
+    ----------
+    line : string
+        A line from ALARA output.txt
+
+    Returns
+    -------
+    decay_times : array of string
+        Array of decay times.
+    """
+    ls = line.strip().split()
+    decay_times = ['shutdown']
+    for i in range(2, len(ls), 2):
+        decay_times.append(''.join([ls[i], ' ', ls[i+1]]))
```

Comment:
can't you just add strings?
```suggestion
        decay_times.append(ls[i] + ' ' + ls[i+1])
```